### DISCUSSION TITLE
Tables in tabs

### DISCUSSION THREAD

**Discussion by jamest-nz** [filamentphp/filament#2539] [Help]
Hi, as someone who has used Nova on several projects I'm loving Filament so far :-)

My project is based around the Admin panel, and I have a Custom Resource Page that renders a Form in its view. In this form's schema is a "Tabs", and inside one of these tabs I'd like to render a custom layout component.

I've got this working as long as I put only normal blade content in my custom component's view template. The custom content shows up in the tab just fine.

Where I go wrong is trying to put a Table into that custom view component. I receive the error:

	Property [$table] not found on component [...]
	
But the component listed is the overall Resource Page component, rather than my inner custom layout component (the Resource Page itself does not have a Table on it, just my custom layout component). I guess the templates are just being merged at the blade level, whereas I'd need to be doing something at the livewire level for this to work.

Is there any way to render a Table inside of a Tab using a custom view like this? I see from a recent discussion (https://github.com/laravel-filament/filament/discussions/2329#discussioncomment-2713436) that one way to get multiple Tables on a single page is to put each table on a separate view component, then render these views on the page in livewire.

I'm not sure how to do that in the case of Tabs, or if it even applies. Any help would be much appreciated!

I think my other alternative would be to have a single Resource Page with multiple Tables on it as above, and implement the UI "tabs" aspect separately myself in CSS.

Cheers

**Comment by danharrin** [comment]
Ah, you're adding the table builder to a view component instead of a Livewire component?

**Comment by jamest-nz** [comment]
Thanks for your reply. Yep I think that's what I've done (perhaps the wrong thing, I'm busy learning Livewire at the same time as Filament!)

Here's what I did: The overall page (containing the Tabs) is a Resource Page made with "artisan make:filament-page" and associated to the resource as a "custom" page, which all works well. My next goal is to have Tables (or RelationManagers) inside several of those Tabs, so I figured I'd run "artisan make make:form-layout" to produce a custom form component to contain the table (putting implements HasTable and use InteractsWithTable on the form component). Generally this had been working well to get arbitrary content into the Tabs, but breaks if I put a table on it.

If the proper way is for me to make a Livewire component containing the Table, then put this into the Tabs (instead of a component made with "make:filament-page") can you point me in the direction of the docs or some code? Sorry if it's a basic thing I've missed!

Thanks again for your help on this.

**Reply by danharrin** [reply]
Yeah, so the InteractsWithTable trait needs to be on a Livewire component, which is different from a form component. Form components are actually Blade view components, not Livewire components.

You will be able to render the Livewire component in the Blade view of your form component, using `@livewire('component-name')`

**Reply by danharrin** [reply]
One thing to note - filters & action forms will not work on tables within the form. This is because HTML does not allow nested `<form>` elements.

**Reply by jamest-nz** [reply]
Ah, that makes sense, really appreciate your help on this!
